packages/databricks-vscode/src/bundle/models/BundleVariableModel.ts

```diff
@@ -0,0 +1,236 @@
+import {BundleSchema} from "../types";
+import {readFile} from "fs/promises";
+import * as os from "os";
+import {NamedLogger} from "@databricks/databricks-sdk/dist/logging";
+import {Loggers} from "../../logger";
+import {onError} from "../../utils/onErrorDecorator";
+import {BundleValidateModel} from "./BundleValidateModel";
+
+export type BundleVariable = Required<BundleSchema>["variables"][string] & {
+    valueInTarget?: string;
+    vscodeOverrideValue?: string;
+    required: boolean;
+};
+
+export type BundleVariableModelState = {
+    variables?: Record<string, BundleVariable>;
+};
+
+export class BundleVariableModel extends BaseModelWithStateCache<BundleVariableModelState> {
+    protected mutex: Mutex = new Mutex();
+    private target: string | undefined;
+    private overrideFileWatcher: FileSystemWatcher | undefined;
+    constructor(
+        private readonly configModel: ConfigModel,
+        private readonly bundleValidateModel: BundleValidateModel,
+        private readonly workspaceRoot: Uri
+    ) {
+        super();
+        this.disposables.push(
+            this.configModel.onDidChangeKey("validateConfig")(async () => {
+                await this.stateCache.refresh();
+            }),
+            this.configModel.onDidChangeKey("preValidateConfig")(async () => {
+                await this.stateCache.refresh();
+            }),
+            this.configModel.onDidChangeTarget(async () => {
+                this.setTarget(this.configModel.target);
+            })
+        );
+    }
+
+    setTarget(target: string | undefined) {
+        if (this.target === target) {
+            return;
+        }
+
+        this.target = target;
+
+        if (this.target === undefined) {
+            this.overrideFileWatcher?.dispose();
+            this.overrideFileWatcher = undefined;
+            return;
+        }
+
+        this.overrideFileWatcher = workspace.createFileSystemWatcher(
+            this.bundleVariableFilePath!.fsPath
+        );
+
+        this.disposables.push(
+            this.overrideFileWatcher,
+            this.overrideFileWatcher.onDidChange(async () => {
+                await this.stateCache.refresh();
+                await this.bundleValidateModel.refresh();
+            }),
+            this.overrideFileWatcher.onDidCreate(async () => {
+                await this.stateCache.refresh();
+                await this.bundleValidateModel.refresh();
+            }),
+            this.overrideFileWatcher.onDidDelete(async () => {
+                await this.stateCache.refresh();
+                await this.bundleValidateModel.refresh();
+            })
+        );
+    }
+
+    get bundleVariableFilePath() {
+        const target = this.configModel.target;
+        if (target === undefined) {
+            return undefined;
+        }
+
+        return Uri.joinPath(
+            this.workspaceRoot,
+            ".databricks",
+            "bundle",
+            target,
+            "vscode.bundlevars"
```

Comment:
Have we considered using `json`? Much easier to validate and show proper errors to the users when the format is incorrect. Right now we rely on try/catch block in the `getVariableOverrides`, but when the format is incorrect the error will not be user friendly